src/compiler/checker.ts

```diff
@@ -24419,6 +24419,12 @@ namespace ts {
                     if (isAssignmentOperator((node as BinaryExpression).operatorToken.kind)) {
                         return false;
                     }
+                    if ((node as BinaryExpression).operatorToken.kind === SyntaxKind.CommaToken) {
+                        // A comma operator is SEF if either operand is SEF, e.g. the template argument in
+                        // `The coordinates are ${x.toString(), y, z}`
+                        // contains an illegally SEF expression at 'y' (the left side of the outer comma whose right operand is 'z')
+                        return isSideEffectFree((node as BinaryExpression).left) || isSideEffectFree((node as BinaryExpression).right);

```

Comment:
So this is less `isSideEffectFree` and more `containsSideEffectFree` now, right? Since technically the comma operator itself would only actually be side-effect free if everything it executed was free of side effects?

Reply:
I don't think so. `f() + 4` *contains* a non-side-effect-free expression, but is side-effect-free due to the addition operator itself not being something that can produce side effects*

Upon thinking about this more, I think the correct fix is just to return `true` if the operator in a binary expression isn't an assignment operator. The expression `[(f() + 4, 2)]` is equally wrong as `[(f() + 4, 2)]`; the comma is a red herring. I'll try this on Monday.

<sub>* let's continue to pretend `valueOf` doesn't exist, per convention</sub>